### ISSUE TITLE
EA FC 24 stuck at the startup

### ISSUE THREAD

**Issue by olus85** [yuzu-emu/yuzu#11634]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Affected Build(s)

All Builds, for example Early Access 3900

### Description of Issue

Game keeps loading at the ea startup screen.
Everyone has this problem, see here: https://www.reddit.com/r/yuzu/comments/16u7dit/ea_fc_24_stuck_at_the_startup/

### Expected Behavior

just pass the loading screen

### Reproduction Steps

just start the game

### Log File

[message.txt](https://github.com/yuzu-emu/yuzu/files/12766577/message.txt)

### System Configuration

CPU: all cpus
GPU/Driver: doesnt matter
RAM: no problem here
OS: all systems

**Comment by FernandoS27** [comment]
Does it boot on single core by any chance?

**Comment by olus85** [comment]
i dont know, sorry. log is from another user. i can test tonight with a steam deck.

**Comment by Rafaelfcosta** [comment]
> Does it boot on single core by any chance?

Tried here, the game dont even boot, stuck on yuzu loading screen

**Comment by FernandoS27** [comment]
Hmm interesting.

**Comment by Squall-Leonhart** [comment]
Believe this title has the first use of Denuvo for switch.

**Comment by dvm23** [comment]
> Believe this title has the first use of Denuvo for switch.

So can this be prevented?

**Comment by FearlessTobi** [comment]
@Squall-Leonhart Do you have any hard evidence for this? Please only make such statements when there is no doubt about it.

**Comment by FernandoS27** [comment]
@FearlessTobi Relax, it's fine to make such an assumption considering no switch emulator can boot it and there are no missing services reported. It also doesn't work in SC.

So it's definitely a possibility.

**Comment by Squall-Leonhart** [comment]
> @Squall-Leonhart Do you have any hard evidence for this? Please only make such statements when there is no doubt about it.

It uses a stream container that has online interactivity during the boot process, which conveys information from EnvironmentService including environmentIdentifier and devicePlatform

**Comment by olus85** [comment]
this looks like a fix:
https://github.com/masagrator/SaltyNX/releases/tag/0.7.0

**Comment by masagrator** [comment]
If I can add a wild guess why this can happen, it may because FC 24 relies on custom implementation of some of NVN stuff. 
Like instead of using nvnWindowBuilderSetTextures they are writing directly to nvnWindowBuilder structure which is not possible with SDK alone, so they reverse engineered some stuff.

```cpp
if ( nvnPrivate0 )
  {
    *(_DWORD *)&builder.reserved[16] = v5;
    *(_QWORD *)&builder.reserved[40] = (char *)&v25 - ((8 * v5 + 15) & 0xFFFFFFFF0LL);
  }
else
  {
    nvn::pfncpp_nvnWindowBuilderSetTextures(&builder, v5, v7);
  }
  ```
  nvnPrivate0 is set, so nvn::pfncpp_nvnWindowBuilderSetTextures is never used.

**Comment by FernandoS27** [comment]
> If I can add a wild guess why this can happen, it may because FC 24 relies on custom implementation of some of NVN stuff. Like instead of using nvnWindowBuilderSetTextures they are writing directly to nvnWindowBuilder structure which is not possible with nnSDK alone, so they reverse engineered some stuff.
> 
> ```c++
> if ( nvnPrivate0 )
>   {
>     *(_DWORD *)&builder.reserved[16] = v5;
>     *(_QWORD *)&builder.reserved[40] = (char *)&v25 - ((8 * v5 + 15) & 0xFFFFFFFF0LL);
>   }
> else
>   {
>     nvn::pfncpp_nvnWindowBuilderSetTextures(&builder, v5, v7);
>   }
> ```
> 
> nvnPrivate0 is set, so nvn::pfncpp_nvnWindowBuilderSetTextures is never used. This is my first time seeing a game that writes directly to nvnWindowBuilder struct.

It shouldn't matter our GPU emulation is LLE.

**Comment by pr030880** [comment]
I can confirm. When I start EA SPORTS FC 24 with latest Update 1.542d45 it loads into the white FC 24 loading screen and it is stuck there forever (waited >30 mins).

**Comment by komputersbest5** [comment]
People have any news when the game will work..?

**Comment by FernandoS27** [comment]
> If I can add a wild guess why this can happen, it may because FC 24 relies on custom implementation of some of NVN stuff. Like instead of using nvnWindowBuilderSetTextures they are writing directly to nvnWindowBuilder structure which is not possible with nnSDK alone, so they reverse engineered some stuff.
> 
> ```c++
> if ( nvnPrivate0 )
>   {
>     *(_DWORD *)&builder.reserved[16] = v5;
>     *(_QWORD *)&builder.reserved[40] = (char *)&v25 - ((8 * v5 + 15) & 0xFFFFFFFF0LL);
>   }
> else
>   {
>     nvn::pfncpp_nvnWindowBuilderSetTextures(&builder, v5, v7);
>   }
> ```
> 
> nvnPrivate0 is set, so nvn::pfncpp_nvnWindowBuilderSetTextures is never used. This is my first time seeing a game that writes directly to nvnWindowBuilder struct.

Actually I dismissed this too quickly. I just had a general feeling it could be something else but many new games seem to be playing with nvflinger.

**Comment by GNS7** [comment]
Any news on this?

**Comment by komputersbest5** [comment]
No does not work so far..

**Comment by furkanonver** [comment]
still any update ?

**Comment by BarbieriGiuseppe** [comment]
any news?

**Comment by komputersbest5** [comment]
https://komputers-best.ru/

**Comment by komputersbest5** [comment]
The problem is in the emulator, yuzu tech support makes excuses, allegedly does not want to deal with it, in fact they just do not know how to solve this problem..

**Comment by FernandoS27** [comment]
Yes the problem is in the emulator as the switch can run it unless it's verifying online. The game now is able to make the window so the original problem that was stated is no longer the case.

There's a different issue which may not be possible to fix anytime soon or ever if my suspicions about denuvo are confirmed. Anti emulation techniques are normally only fixable through modding. They are meant to abuse mechanisms that are extremely hard to emulate.
